src/LayoutManager/LayoutManager.ts

```diff
@@ -55,51 +55,71 @@ export class LayoutManager {
     this.onBeforeLayout(strictContext);
 
     const layoutResult = this.getLayoutResult(strictContext);
-    layoutResult && this.commitLayout(strictContext, layoutResult);
+    if (layoutResult) {
+      this.commitLayout(strictContext, layoutResult);
+    }
 
     this.onAfterLayout(strictContext, layoutResult);
     this._prevLayoutStrategy = strictContext.strategy;
   }
 
   /**
-   * subscribe to object layout triggers
+   * Attach handlers for events that we know will invalidate the layout when
+   * performed on child objects ( general transforms ).
+   * Returns the disposers for later unsubscribing and cleanup
+   * @param {FabricObject} object
+   * @param {RegistrationContext & Partial<StrictLayoutContext>} context
+   * @returns {VoidFunction[]} disposers remove the handlers
+   */
+  protected attachHandlers(
+    object: FabricObject,
+    context: RegistrationContext & Partial<StrictLayoutContext>
+  ): VoidFunction[] {
+    const { target } = context;
+    return (
+      [
+        'moving',
+        'resizing',
+        'rotating',
+        'scaling',
+        'skewing',
+        'changed',
+        'modifyPoly',
+        'modified',
+      ] as (TModificationEvents & 'modified')[]
+    ).map((key) =>
+      object.on(key, (e) =>
+        this.performLayout(
+          key === 'modified'
+            ? {
+                type: LAYOUT_TYPE_OBJECT_MODIFIED,
+                trigger: key,
+                e,
+                target,
+              }
+            : {
+                type: LAYOUT_TYPE_OBJECT_MODIFYING,
+                trigger: key,
+                e,
+                target,
+              }
+        )
+      )
+    );
+  }
+
+  /**
+   * Subscribe an object to transform events that will trigger a layout change on the parent
+   * This is important only for interactive groups.
+   * @param object
+   * @param context
    */
   protected subscribe(
     object: FabricObject,
     context: RegistrationContext & Partial<StrictLayoutContext>
```

Comment:
I think we should change the type for all subscription methods
```suggestion
    context: RegistrationContext
```